CMS-Enterprise/sbom-harbor
# refactor: Simplify `Package` data model

## Summary

This PR contains changes to the data model and associated logic.

### Added

- `Sbom` has new fields that represent the NTIA minimum elements of an Sbom

### Changed

- `Dependency` and `Purl` structs that were useful during the design and debug phase have now be merged into the single `Package` struct.
- `Vulnerability` is now a standalone collection at the data store level, rather than an embedded object.

### Deprecated

N/A

### Removed

- `Dependency` struct
- `Purl` struct
- `API_VERSION` is no longer part of the `SbomProviderKind:Snyk` enum.

### Fixed

N/A

## How to test

- Start the local development environment
- Delete any existing Mongo Collections - this data model is not backward compatible
- `cargo build` from the root of the repo
- `./target/debug/harbor-cli sbom --debug -p snyk` to generate SBOMs
- Review the data in the `Sbom`, `Package`,  `Unsupported` and `Task` collections.
- Verify SBOM files have been saved to `/tmp/harbor-debug/sboms`.
- `./target/debug/harbor-cli enrich --debug -p snyk` to generate Vulnerabilties
- Review the data in the `Vulnerability` collection.
